packages/clerk-js/src/core/modules/commerce/CommerceBilling.ts

```
@@ -29,15 +28,22 @@ import {
 } from '../../resources/internal';
 
 export class CommerceBilling implements CommerceBillingNamespace {
-  getPlans = async (params?: GetPlansParams): Promise<CommercePlanResource[]> => {
-    const { data: products } = (await BaseResource._fetch({
-      path: `/commerce/products`,
+  getPlans = async (params?: GetPlansParams): Promise<ClerkPaginatedResponse<CommercePlanResource>> => {
+    const safeParams = Object.assign({}, params);
+    (safeParams as any).payer_type = safeParams.for || 'user';
+    delete safeParams.for;
```

Biome (1.9.4)

[error] Avoid the delete operator which can impact performance.

Unsafe fix: Use an undefined assignment instead.


(lint/performance/noDelete)